lib/queue.js

```diff
@@ -355,6 +440,9 @@ Queue.prototype._finishJob = function (err, data, job, cb) {
   };
 
   if (err) {
+    job.options.attemptsMade += 1;
+    job.options.stacktraces.unshift(err.stack);
```

Comment:
TIL `Error.prototype.stack` is non-standard: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack